lib/modules/platform/gerrit/client.spec.ts

```diff
@@ -0,0 +1,379 @@
+import { Fixtures } from '../../../../test/fixtures';
+import * as httpMock from '../../../../test/http-mock';
+import { partial } from '../../../../test/util';
+import { REPOSITORY_ARCHIVED } from '../../../constants/error-messages';
+import { setBaseUrl } from '../../../util/http/gerrit';
+import { GerritClient } from './client';
+import type {
+  GerritChange,
+  GerritChangeMessageInfo,
+  GerritMergeableInfo,
+} from './types';
+
+const gerritEndpointUrl = 'https://dev.gerrit.com/renovate/';
+const jsonResultHeader = { 'content-type': 'application/json;charset=utf-8' };
+
+describe('modules/platform/gerrit/client', () => {
+  const client = new GerritClient();
+
+  beforeAll(() => {
+    setBaseUrl(gerritEndpointUrl);
+  });
+
+  describe('getRepos()', () => {
+    it('returns repos', async () => {
+      httpMock
+        .scope(gerritEndpointUrl)
+        .get('/a/projects/?type=CODE&state=ACTIVE')
+        .reply(
+          200,
+          gerritRestResponse({
+            repo1: { id: 'repo1', state: 'ACTIVE' },
+            repo2: { id: 'repo2', state: 'ACTIVE' },
+          }),
+          jsonResultHeader
+        );
+      expect(await client.getRepos()).toEqual(['repo1', 'repo2']);
+    });
+  });
+
+  describe('getProjectInfo()', () => {
+    it('inactive', () => {
+      httpMock
+        .scope(gerritEndpointUrl)
+        .get('/a/projects/test%2Frepo')
+        .reply(
+          200,
+          gerritRestResponse({
+            id: 'repo1',
+            name: 'test-repo',
+            state: 'READ_ONLY',
+          }),
+          jsonResultHeader
+        );
+      return expect(client.getProjectInfo('test/repo')).rejects.toThrow(
```

Comment:
use await instead of return

Reply:
Okay, but is there any benefit of this syntax?

Reply:
looks cleaner and better stack traces on errors.